mne/forward/_make_forward.py

```diff
@@ -605,6 +612,123 @@ def make_forward_solution(info, trans, src, bem, fname=None, meg=True,
+        time courses.
+    fwd : instance of Forward
+        The forward solution.
+
+    See Also
+    --------
+    mne.simulation.simulate_evoked
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    # Make copies to avoid mangling original dipole
+    times = dipole.times.copy()
+    pos = dipole.pos.copy()
+    amplitude = dipole.amplitude.copy()
+    ori = dipole.ori.copy()
+
+    # Convert positions to discrete source space (allows duplicate rr & nn)
+    # NB information about dipole orientation enters here, then no more
+    sources = dict(rr=pos, nn=ori)
+    # Dipole objects must be in the head frame
+    sp = _make_discrete_source_space(sources, coord_frame='head')
+    src = SourceSpaces([sp])  # dict with working_dir, command_line not nec
+
+    # Forward operator created for channels in info (use pick_info to restrict)
+    # Use defaults for most params, including min_dist
+    fwd = make_forward_solution(info, trans, src, bem, fname=None,
+                                n_jobs=n_jobs, verbose=verbose)
+    # Convert from free orientations to fixed (in-place)
+    convert_forward_solution(fwd, surf_ori=False, force_fixed=True,
+                             copy=False, verbose=None)
+
+    # Check for omissions due to proximity to inner skull in
+    # make_forward_solution, which will result in an exception
+    if fwd['src'][0]['nuse'] != len(pos):
+        inuse = fwd['src'][0]['inuse'].astype(np.bool)
+        head = ('The following dipoles are outside the inner skull boundary')
+        msg = len(head)*'#' + '\n' + head + '\n'
+        for (t, pos) in zip(times[np.logical_not(inuse)],
+                            pos[np.logical_not(inuse)]):
+            msg += '    t={:.0f} ms, pos=({:.0f}, {:.0f}, {:.0f}) mm\n'.\
+                format(t*1000., pos[0]*1000., pos[1]*1000., pos[2]*1000.)
+        msg += len(head)*'#'
+        print(msg)
+        raise ValueError('One or more dipoles outside the inner skull')
+
+    # multiple dipoles (rr and nn) per time instant allowed
+    # uneven sampling in time returns list
+    timepoints = np.unique(times)
+    if len(timepoints) > 1:
+        tdiff = np.diff(timepoints)
+        if not np.allclose(tdiff, tdiff[0]):
+            warn('Unique time points of dipoles unevenly spaced: returned '
+                 'stc will be a list, one for each time point.')
+            tstep = -1.0
+        else:
+            tstep = tdiff[0]
+    elif len(timepoints) == 1:
+        tstep = 0.001
+
+    # Build the data matrix, essentially a block-diagonal with
+    # n_rows: number of dipoles in total (dipole.amplitudes)
+    # n_cols: number of unique time points in dipole.times
+    # amplitude with identical value of times go together in one col (others=0)
+    data = np.zeros((len(amplitude), len(timepoints)))  # (n_d, n_t)
+    row = 0
+    for tpind, tp in enumerate(timepoints):
+        amp = amplitude[in1d(times, tp)]
+        data[row:row+len(amp), tpind] = amp
```

Comment:
not surprising there -- spaces between operators is not universally agreed upon (and e.g. scipy doesn't do it), but as a convention we follow it